### DISCUSSION TITLE
Problems with the Multiranger-Deck

### DISCUSSION THREAD

**Discussion by jmubo** [bitcraze/discussions#675] [Q&A]
Hi everybody, 
i am experiencing problems with the Multiranger-Deck that i am not able to solve right now. When i connect the Multiranger-Deck to the Crazyflie, the Crazyflie starts the system test over and over again when powered up. When i use the CFClient to conntect to the Crazyflie, i get the following output:

SYS: Crazyflie 2.1 is up and running!
SYS: Build 67:9956e68e10ba (2023.02 +67) MODIFIED
SYS: I am 0x38353530353947090041002D and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_CORE: 2 deck(s) found
DECK_CORE: Calling INIT on driver bcMultiranger for deck 0
DECK_CORE: Calling INIT on driver bcZRanger2 for deck 1
ZR2: Z-down sensor [OK]
IMU: BMI088: Using I2C interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Complementary (1) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushed motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2022.05
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
MR: Init front sensor [OK]
MR: Init back sensor [OK]
MR: Init up sensor [OK]
SYS: ----------------------------
SYS: Crazyflie 2.1 is up and running!
SYS: Build 67:9956e68e10ba (2023.02 +67) MODIFIED
OW: Cmd 0x22 timeout.
DECK_INFO: Reading deck nr:0 [FAILED]. No driver will be initialized!
DECK_CORE: 0 deck(s) found
IMU: BMI088: Using I2C interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Complementary (1) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushed motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2022.05
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
SYS: The system resumed after a failed assert [WARNING]
SYS: Assert failed at ../src/drivers/src/uart_syslink.c:549
SYS: cfAssertNormalStart [FAIL]

To me it seems that the Crazyflie starts for one time, detects both connected decks (Multiranger and Z-Ranger V2), but the systemtest of the Multiranger fails (probably left or right sensor). The system reboots and starts the Crazyflie without any decks. 
Does that behavior indicate a broken Deck or sensor? The Deck is brandnew and tested for the first time.

I used the freshly updated firmware for my testes. However, the problem occurs with all firmware versions i tested so far.

I would appreciate any kind of help.

Best,
jmubo

**Comment by ataffanel** [comment]
Hi,

The main problem here seems to be these two lines;
```
OW: Cmd 0x22 timeout.
DECK_INFO: Reading deck nr:0 [FAILED]. No driver will be initialized!
```

Decks have a OneWire memory that is used to detect and initialize them, in your case this one wire memory cannot be read for some reason.

The most likely case is that the connection is not done properly. You can try to connect only the multi-ranger deck with long pins sticking up to see if you can get a better physical connection and if it help.

**Reply by jmubo** [reply]
Hi,

thanks for your reply. I noticed just now that i did not answer. 
It turned out to be a bad soldering of one of the indivual sensors of the multiranger (probably after transportation). 
We resoldered the sensor and now everything works like a charm.
